### PR TITLE
1. Add inline fast cross referencing

### PR DESCRIPTION
**Pull request by chhaymenghong**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -79,19 +79,49 @@ module.exports = function(grunt) {
     }
   }
 
+  var combineCustomJS = {
```
Comment (shakeh) [inline Gruntfile.js]: glad you're adding this!

Comment (chhaymenghong) [reply Gruntfile.js]: haha. sorry for the inconveniences. I also had enough of it.

**Comment by chhaymenghong** [comment]
added esc handler. Good idea!

**Comment by dlamoris** [comment]
doesn't seem to work on firefox, works in chrome

**Comment by dlamoris** [comment]
seems like on firefox the dropdown is too far down, sometimes it disappears off screen.
also on escape can it remove the placeholder span?

File: src/directives/mmsCkEditor.js [hunk 2]
```diff
@@ -455,33 +389,18 @@ function mmsCkeditor(CacheService, ElementService, UtilsService, ViewService, UR
...
             });
-            var deb = _.debounce(function(e) {
-                update();
-            }, 1000);
-            instance.on( 'change', deb);
-            instance.on( 'afterCommandExec', deb);
-            instance.on( 'resize', deb);
-            instance.on( 'destroy', deb);
+
+            instance.on( 'change', _deb());
```
[the remark this answers is not in the document]
Comment (chhaymenghong) [reply src/directives/mmsCkEditor.js]: ah you are right. this is wrong. my intention was to clean up code, but I forgot that _.debounce would return a separate fn for each of this event type. I will revert it. thx you

**Comment by chhaymenghong** [comment]
I intentionally keep the mention state around even after "esc" is pressed and I think we should do that because 1. it follows "Slack" mentioning feature and 2. for a situation where users change their minds and actually want to do fast cf. By not removing the state, it can be reactivated.

**Comment by chhaymenghong** [comment]
yeah.When there is a scrollbar in the ckeditor and it is scrolled from the top of the editor, the fast cf dropdown positioning is messed up because I was using offsetTop instead getBoundingClientRect to get top position relative to viewport inside the ckeditor's iframe. it is fixed now.

**Comment by chhaymenghong** [comment]
Per discussion, unwrap contents on esc.

**Comment by shakeh** [comment]
@chhaymenghong was just about to merge this but noticed sometimes i have to do the '@' twice before i get it to show up. have you had that happen?

**Comment by chhaymenghong** [comment]
i think you just gotta make sure that shift and @ are clicked at about the same time.

**Pull request merged**